### PR TITLE
Rover: set MIS_DONE_BEHAVE to LOITER in boats

### PR DESCRIPTION
**Pull request by shiv-tyagi**
This fixes the issue #19305. I have tested this and got expected result. Here's a screenshot.
test

### CODE REVIEW

File: Rover/system.cpp [hunk 1]
```diff
@@ -136,6 +136,11 @@ void Rover::init_ardupilot()
 
     rover.g2.sailboat.init();
 
+    // boat should loiter after completing a mission to avoid drifting off
+    if (is_boat()) {
+        AP_Param::set_default_by_name("MIS_DONE_BEHAVE", 1);
```
Comment (IamPete1) [inline Rover/system.cpp:141]:
```suggestion
        g2.mis_done_behave.set_default(MIS_DONE_BEHAVE_LOITER);
```
saves the string lookup and used the enum value to set.

Comment (rmackay9) [reply Rover/system.cpp:141]: Ah right, because the parameter is visible to the vehicle class this works.  Seems like a good suggestion to me.

Comment (shiv-tyagi) [reply Rover/system.cpp:141]: Ok. I'll change it to use `set_default`. 
The  `Mis_Done_Behave` is private to `ModeAuto` class. Should I make it public?

Comment (rmackay9) [reply Rover/system.cpp:141]: @shiv-tyagi, yes, making the enum public seems OK to me. txs

Comment (shiv-tyagi) [reply Rover/system.cpp]: Done!

**Review by rmackay9 — APPROVED** [review]

**Comment by rmackay9** [comment]
Thanks!  I've been looking forward to this fix

**Pull request merged**